AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/MutationSync/OutgoingMutationQueue/OutgoingMutationQueue.swift

```diff
@@ -241,10 +245,37 @@ final class OutgoingMutationQueue: OutgoingMutationQueueBehavior {
                 self.log.verbose("mutationEvent deleted successfully")
             }
 
+            self.dispatchOutboxStatusEvent(isEmpty: true)
             self.stateMachine.notify(action: .processedEvent)
         }
     }
 
+    private func checkStatus() -> Bool {
+        var isEmpty = true
+        let fields = MutationEvent.keys
+        let predicate = fields.inProcess == false || fields.inProcess == nil
+
+        storageAdapter.query(MutationEvent.self,
+                             predicate: predicate,
+                             sort: nil,
+                             paginationInput: nil) { result in
+            switch result {
+            case .success(let events):
+                isEmpty = events.isEmpty
+            case .failure(let error):
+                log.error("Error quering mutation events: \(error)")
+            }
+        }
+        return isEmpty
```

Comment:
storageAdapter completion closure `{ result in ...` comes back asynchronously, so you will always return isEmpty true at line 269

Reply:
synced offline on this and it appears storageAdapter calls the completion handler before returning, making it essentially a sycrhonous operation, however we'll still code against it like it is async